scripts/cairo_programs_verifier.sh

```diff
@@ -4,39 +4,67 @@ GREEN='\033[1;32m'
 RED='\033[1;31m'
 NC='\033[0m'
 
-# flag to check if any errors were encountered
-has_errors=false
-
-# function to process directory
-process_directory() {
-  for dir in "$1"/*
-  do
-    if [ -d "$dir" ]; then
-      echo "Processing $dir"
-      (cd "$dir" && scarb build 0>/dev/null 1> error.log && scarb fmt -c 0>/dev/null 1>> error.log && scarb test 0>/dev/null 1>> error.log)
-      if [ $? -ne 0 ]; then
-        has_errors=true
-        echo "Error while processing $dir"
-        cat "$dir/error.log"
-      fi
-      rm "$dir/error.log"
+# root directory of the repository
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+error_file=$(mktemp)
+
+# function to list modified cairo files
+list_modified_cairo_files() {
+    git diff --name-only main...HEAD -- listings | grep -E 'listings/ch.*/*.cairo$'
```

Comment:
Less restrictive is ok, like `listings/ch.*/*`, so even if there's only a config change `Scarb.toml` the script still verify the listing.

Reply:
Also we need to be sure to compare against the upstream `NethermindEth/StarknetByExample` main branch and not the local main branch.